dulwich/tests/test_index.py

```diff
@@ -369,7 +369,10 @@ def test_nonempty(self):
                 sorted(os.listdir(os.path.join(repo.path, 'c'))))
 
     def test_norewrite(self):
-        sync = getattr(os, 'sync', lambda: os.system('sync'))
+        if sys.platform != 'win32':
```

Comment:
Hmm, skipping the test if os.sync is not present will also have it skipped when using python2. I can check if sync is a system command of course, but that's going to be a pretty complicated check for finding out whether to skip the test..

Just checking if os.sync exists is most straight-forward it seems, but feels like throwing out the baby with the bathwater..